src/state/update.go

```diff
@@ -12,6 +12,11 @@ import (
 // EnterNode must be called upon entering new node to update current state.
 func EnterNode(st *meta.ClassParseState, n walker.Walkable) {
 	switch n := n.(type) {
+	case *stmt.Function:
+		st.CurrentFunction = meta.NameNodeToString(n.FunctionName)
+	case *stmt.ClassMethod:
+		st.CurrentFunction = meta.NameNodeToString(n.MethodName)
```

Comment:
// NameNodeToString converts nodes of *name.Name, *name.FullyQualified and *node.Identifier to string.
// This function is a helper function to aid printing function names, not for actual code analysis.